mautic/mautic
# Add new report data source 'Focus stats'

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🔴 
| New feature/enhancement? (use the a.x branch)      | 🟢
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🟢 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #... 



## Description

This PR adds an extra report data source called 'Focus Stats'.

![image](https://github.com/user-attachments/assets/139e7f50-1e22-4ca9-aad0-44b58e36f1fd)

There one can select the columns: name, url, interaction, page hits, unique hits, description, focus type, style

![image](https://github.com/user-attachments/assets/8f94db9a-469b-4395-b0d7-4ebd0faf527d)

The report is meant to show statistics for page hits and unique hits for both your focus item views and clicks.

![image](https://github.com/user-attachments/assets/6897dd12-6095-479c-9a96-e11c8cd35b9a)






---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create some focus stats data by:
         1. Creating multiple focus items
         2. Adding focus items to a landing page
         3. Open the landing pages in different incognito browser windows multiple times to add some views / views+clicks
3. Create a report with data source ‚focus stats‘ and select all columns
4. Check if all the data in the report corresponds to your behavior from 2.3 (you can also check in your database via the tables focus, focus_stats, channel_url_trackables, page_redirects)
